### PR TITLE
Return error if BOOL8 column-type is used with integers-to-hex

### LINKED ISSUE #14232: [DOC] Incorrect value for integers_to_hex reported in the function docs
## Report incorrect documentation
The value 123 should map to '7B' not '4D2' in the function docs for integers_to_hex. 

**Location of incorrect documentation**
https://github.com/rapidsai/cudf/blob/23d24d43fac8615166c38231f13fe8751a8aec42/cpp/include/cudf/strings/convert/convert_integers.hpp#L204

**Describe the problems or issues found in the documentation**
It should be fixed to the correct hex value

**Steps taken to verify documentation is incorrect**
Ran the method `integers_to_hex`

---

## Report needed documentation

**Report needed documentation**
A clear and concise description of what documentation you believe it is needed and why.

**Describe the documentation you'd like**

**Steps taken to search for needed documentation**
List any steps you have taken:

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Removes support to convert BOOL8 column-type to hex using `cudf::strings::integers_to_hex`.
Also fixed other integer to string conversions to remove this unsupported type.
Added gtests to verify an error is thrown for this case.

Closes #14232 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Review by bdice — APPROVED** [review]

**Review by ttnghia — APPROVED** [review]
This PR makes me think of something similar to `cudf::test::IntegralTypesNotBool` in the test module. Should we have something like this:
```
namespace cudf {
template<typename T>
bool constexpr is_integral_not_bool() {}
....
```

File: cpp/src/strings/convert/convert_hex.cu [hunk 1]
```diff
@@ -197,7 +194,8 @@ struct dispatch_integers_to_hex_fn {
   }
   // non-integral types throw an exception
   template <typename T, typename... Args>
-  std::enable_if_t<not std::is_integral_v<T>, std::unique_ptr<column>> operator()(Args...) const
+  std::enable_if_t<not std::is_integral_v<T> or std::is_same_v<T, bool>, std::unique_ptr<column>>
```
Comment (karthikeyann) [inline cpp/src/strings/convert/convert_hex.cu:197]: nit: this seems like a common trait. consider adding it to traits.hpp

Comment (davidwendt) [reply cpp/src/strings/convert/convert_hex.cu:197]: There is an `cudf::is_index_type()` trait check that is equivalent to this but I felt the current code more accurately communicated the intention here.

File: cpp/src/strings/convert/convert_hex.cu [hunk 2]
```diff
@@ -107,20 +109,13 @@ struct dispatch_hex_to_integers_fn {
   }
   // non-integral types throw an exception
   template <typename T, typename... Args>
```
[the remark this answers is not in the document]
Comment (davidwendt) [reply cpp/src/strings/convert/convert_hex.cu:111]: What is wrong with the comment? Would `non-integer` be more clear?

File: cpp/src/strings/convert/convert_hex.cu (hunk 1 above)
Comment (ttnghia) [reply cpp/src/strings/convert/convert_hex.cu:197]: Then `is_integral_not_bool` should clearly communicate its intention.

**Review by karthikeyann — APPROVED** [review]

File: cpp/src/strings/convert/convert_hex.cu (hunk 2 above)
[the remark this answers is not in the document]
Comment (karthikeyann) [reply cpp/src/strings/convert/convert_hex.cu:110]: Yes. non-integral and bool. (which would be non-integer). It is safe to ignore this. 
This PR is good to merge.

File: cpp/include/cudf/utilities/traits.hpp [hunk 3]
```diff
@@ -279,6 +279,30 @@ constexpr inline bool is_integral()
...
 
+/**
+ * @brief Indicates whether the type `T` is an integral type but not bool type.
+ *
+ * @tparam T  The type to verify
+ * @return true `T` is integral but not bool
+ * @return false  `T` is not integral or is bool
+ */
+template <typename T>
+constexpr inline bool is_integral_not_bool()
```
Comment (mythrocks) [inline cpp/include/cudf/utilities/traits.hpp:290]: Wow, I could have sworn we had this already. (I think I'm confusing this with a type-list in the CUDF tests.)

Thanks for adding this.

**Review by mythrocks — APPROVED** [review]

**Pull request merged**